fog/ocall_oram_storage/trusted/src/lib.rs

```diff
@@ -351,6 +333,16 @@ where
                     dest_meta[idx].copy_from_slice(meta);
                 }
             }
+
+            // Check the last hash with the trusted merkle root storage
+            // This unwrap is valid because if first_treetop_index is zero, then we didn't
+            // enter this `if`
+            let (last_idx, last_hash) = last_hash.expect("should not be empty at this point");
+            assert!(
+                bool::from(self.trusted_merkle_roots[last_idx as usize].ct_eq(&last_hash)),
+                "authentication failed, trusted merkle root mismatch at {}",
+                last_idx
+            );
```

Comment:
@wjuan-mob this is what i thought to do to fix it at first, but the tests are still failing, so this can't be the whole story

lets think about it some more on monday

Reply:
Still working through this. So far my suspicion is that this fix is correct, but that the test is wrong somewhere. I went through and printed out what the merkle root is after "clear" and I got the following which seems to suggest we may have cleared some hashes over zealously: 
```
cargo:warning=First treetop index 16
cargo:warning=trusted merkle root: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], lash_hash: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```